internal/watcher/watcher_plugin.go

```diff
@@ -206,14 +228,36 @@ func (w *Watcher) handleConfigApplyComplete(ctx context.Context, msg *bus.Messag
 			return element == instanceID
 		},
 	)
+
 	w.fileWatcherService.SetEnabled(true)
 }
 
+func (w *Watcher) handleCredentialUpdate(ctx context.Context) {
+	slog.DebugContext(ctx, "Received credential update topic")
+
+	w.watcherMutex.Lock()
+	conn, err := grpc.NewGrpcConnection(ctx, w.agentConfig)
+	if err != nil {
+		slog.ErrorContext(ctx, "Unable to create new grpc connection", "error", err)
+		return
```

Comment:
Lock needs to be unlocked in the error case
```suggestion
                w.watcherMutex.Unock()
		return
```

Reply:
Thanks, added